wordpress-mobile/WordPress-iOS
# Reader: only show seen toggle if seen is supported

Ref: #15355
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/341

This adds a check to the post menu options to show the `Mark as seen` / `Mark as unseen` option based on `isSeenSupported`.

Data model changes:
- The default value of `isSeen` is now `true` to align with Calypso.
- `isSeenSupported` is added to `ReaderPost`.

To test:

The Calypso [change](https://code.a8c.com/D56224) to exclude `is_seen` from older posts is not yet merged, so some posts will not be correct. Since the `read/search` endpoint doesn't return `is_seen`, we can use that to test this logic.

- On any post card in the Reader, tap the options menu button.
  - Verify `Mark as seen` / `Mark as unseen` is present.
- Tap the post card. In the post details, tap the options menu button.
  - Verify `Mark as seen` / `Mark as unseen` is present.
- Tap the search button on the nav bar.
  - Enter a search term.
  - On a post card and post details options menus (same as above), verify `Mark as seen` / `Mark as unseen` is not present.

| Post card | Search card |
|--------|-------|
| ![post_card](https://user-images.githubusercontent.com/1816888/106330298-30239780-6240-11eb-97c2-3c9a0b32bc6d.png) | ![search](https://user-images.githubusercontent.com/1816888/106330317-3a459600-6240-11eb-9cff-10f56e80cd22.png) |

PR submission checklist:
